src/main/java/org/jboss/logmanager/LoggerNode.java

```diff
@@ -103,6 +103,8 @@
      */
     private volatile int effectiveLevel = Logger.INFO_INT;
 
+    private volatile LogRecordPublisher publisher;
```

Comment:
I think it would be better to have one single publisher field on the LogContext instance, so that you can replace it with a single write.

Reply:
So the `LoggerNode` would just use `context.publisher.publish()` then? That makes sense.

Reply:
Yeah.  I think the indirection should be cheap as the LogContext instance would probably stay in cache.